Library/Tracking/KSRAnalytics.swift

```diff
@@ -961,29 +969,27 @@ public final class KSRAnalytics {
     optimizelyProperties _: [String: Any] = [:]
   ) {
     let props = projectProperties(from: project, loggedInUser: self.loggedInUser)
+      .withAllValuesFrom(contextProperties(page: .projectPage))
 
     switch stateType {
     case .fix:
       self.track(
         event: ApprovedEvent.managePledgeButtonClicked.rawValue,
-        page: .projectPage,
-        properties: props.withAllValuesFrom(contextProperties()) // .fixErroredPledge
+        properties: props // .fixErroredPledge
       )
     case .pledge:
       let allProps = props
         .withAllValuesFrom(optimizelyProperties() ?? [:])
-        .withAllValuesFrom(contextProperties(ctaContext: .pledgeInitiate))
+        .withAllValuesFrom(contextProperties(ctaContext: .pledgeInitiate, page: .projectPage))
```

Comment:
You can delete `.projectPage` here because you are assigning it on line 972?

Reply:
If I remove it there 
`contextProperties(ctaContext: .pledgeInitiate)` would reset page to nil because the function is called after you've already set it.

Another option is to explicitly define `contextProperties(page: .projectPage)` in each of the switch case.

Reply:
Ah good point. Can you test this out and see if it works:

`let allProps = contextProperties(ctaContext: .pledgeInitiate).withAllValuesFrom(optimizelyProperties() ?? [:]).withAllValuesFrom(props)`

Reply:
This worked!